src/http-client/index.ts

```diff
@@ -1,7 +1,15 @@
+import Config from '../config';
 import Dispatcher from '../event';
 import TokenManager from '../token-manager';
 import { withTimeout } from '../util/timeout';
-import { HttpClientRequestOptions, RequiresNewTokenFn } from './interfaces';
+import {
+  buildTxnAuthReqOptions,
+  examineForIGTxnAuth,
+  examineForRESTTxnAuth,
+  normalizeIGJSON,
+  normalizeRESTJSON,
+} from './util';
```

Comment:
Rather than writing a bunch of `private` methods on the class, I decided with simple functions within a utility file. This greatly enables testability and keeps the class relatively simple, but it does reveal these methods as "importable". I can implement them as `private` methods if that's the desired approach. If we keep them as external functions, I'd recommend moving the `newTokenRequired` method to be consistent with this convention.